### ISSUE TITLE
Can't install on LolliPop CyanogenMod 12

### ISSUE THREAD

**Issue by andreamoro** [romannurik/dashclock#844]
Just updated my OPO, and I though the new beta would have resolved the problem. I was wrong. I get a "App cannot install" error message.

**Comment by LamerTex** [comment]
Same here, "error -505" and also on a stock nexus4 with Android 5.1

**Comment by Luke1991** [comment]
Same here, commenting to follow

**Comment by RRuszczyk** [comment]
Confirmed here as well.

**Comment by romannurik** [comment]
So, my guess is there's something built into CyanogenMod that uses the DashClock permissions. I'm not a CM user myself so I can't check.

**Comment by RRuszczyk** [comment]
@romannurik I think it's more likely an issue with 5.x.x, as @LamerTex mentions he has an issue with stock android on a Nexus device.

**Comment by LamerTex** [comment]
After a factory reset from twrp (wipe of cache data and dalvik) I'm now able to install it, can somebody else try this?

**Comment by andreamoro** [comment]
Got it. By routing the phone and removing trebuchet you can get the permission problem out of your way

**Comment by romannurik** [comment]
So CyanogenMod bundles a launcher that uses the DashClock permission? Not much I can do about that…

**Comment by andreamoro** [comment]
Well, I think the problem is not Cyanogenmod but the way how Android 5.1 has changed the permissions. I can't really say as I'm just an OP, but I believe that this is happening because some "permissions" have been grouped and now only one app at a time can use the same bundle?

Take my statement with a big pinch of salt.

**Comment by LamerTex** [comment]
I don't know if it's trebuchet the problem, I had the problem and after a factory reset it went away but both times I was with the Google launcher (and trebuchet installed)...

**Comment by DvTonder** [comment]
The bug is in android which does not recognize removed permissions from a apk when it gets upgraded. Trebuchet and cmhome  in cm11 had the permission, it was removed from trebuchet in cm12 but if you upgraded, this removed permission is not removed by Android and as such it thinks trebuchet still has it.  The only solution is to uninstall trebuchet and cmhome completely. You can install the cm12 version of trebuchet afterwards if you really need it.

**Comment by andreamoro** [comment]
Maybe I'm going OT, but I'm not a mobile dev and I'm curious to know. 

How does the ACL and permission system work on Android? I can't really understand how it is possible an app to prevent another to be installed because it has "the same permission". 
Are "the permission" something connected to read, write and something? Because if this is the case, I can think to million of scenarios where more than an app could share permissions.

Thanks

**Comment by DvTonder** [comment]
Apps can declare their own specific permissions.  In versions of android
before L more than one app could define the same permission - in this case
the READ_EXTENSION_DATA permission that dashclock extensions look for in
order to work.  In L, Google changed it such that only one application on
the system can define a permission.  Whatever is the first application that
defines it, no other apps that try to define it will install (the 505
error)

**Comment by globexdesigns** [comment]
@DvTonder How do you uninstall Trebuchet and cmhome? They don't appear in the Apps listing under Settings for me.

**Comment by DvTonder** [comment]
You can use titanium backup to find the two apps In the list and uninstall
them. You will need to temporarily enable root to do so (it's in
development settings)

**Comment by tech189** [comment]
@romannurik With the latest CyanogenMod update, CM 12.1 (Android 5.1.1) I am now able to install Dashclock! :D

**Comment by NewCorn** [comment]
@tech189 HOW? :(
I'm on stock CM 12.1 and drive crazy about this app, which still does not work for me.

**Comment by tech189** [comment]
@NewCorn I don't know... It just works, I just install it from the Play Store and add the widget to my home screen. I'm on CyanogenOS version 12.1-YOG4PAS1N0 if that helps.

**Comment by NewCorn** [comment]
@tech189 Nope, as I mentioned, I run the same CM (not OS, OP users still get the Mod, not the OS) as you, unrooted, and still got this 505 Error.

**Comment by tech189** [comment]
@NewCorn I'm pretty sure I'm running CMOS because my Themes app is different from when I install CM on my rooted Samsung S3, but if you are running the same build, this is a very strange issue. I guess a factory reset would be a sure sign that it doesn't work on Lollipop, I did a factory reset when I first updated to 5.0.2. But doing a factory reset is not fun to do, especially with no Titanium Backup...

**Comment by gypngn** [comment]
emmm…
